src/core/abuse-core.js

```diff
@@ -0,0 +1,196 @@
+              if (_.isEmpty(feedItems)) {
+                throw new Error(
+                  "Feed item not found: " + reports[0].feed_item_id
+                );
+              }
+              return trx("feed_item_reports")
+                .update("is_resolved", true)
+                .where("feed_item_id", feedItems[0])
+                .returning("id");
+            });
+        }
+      })
+      .catch(err => {
+        throw err;
+      });
+  });
+}
+
+/*
+  Fetches all of the reported feed items after a certain date.
+  Also fetches information on the reporter and the feed item itself.
+*/
+function getReportedFeedItems(params) {
+  let whereClause = [];
+  if (params.beforeId) {
+    whereClause.push("feed_item_reports.id < " + params.beforeId + " AND ");
+  }
+  whereClause += "is_resolved = false";
+
+  return knex("feed_item_reports")
+    .count("id")
+    .where("is_resolved", false)
+    .then(number_of_rows => {
+      return knex
+        .from("feed_item_reports")
+        .select(
+          "feed_item_reports.id as report_id",
+          "feed_item_reports.feed_item_id as id",
+          "report_creator_id",
+          "feed_item_reports.created_at as createdAt",
+          "report_description",
+          "is_resolved",
+          "user_id",
+          "location",
+          "feed_items.image_path",
+          "teams.name as team_name",
+          "users.name",
+          "users.id as user_id",
+          "text",
+          "type",
+          "feed_items.is_banned",
+          "hot_score",
+          "teams.city_id",
+          "parent_id"
+        )
+        .innerJoin(
+          "feed_items",
+          "feed_item_reports.feed_item_id",
+          "feed_items.id"
+        )
+        .innerJoin("users", "feed_item_reports.report_creator_id", "users.uuid")
+        .innerJoin("teams", "users.team_id", "teams.id")
+        .whereRaw(whereClause)
+        .limit(20)
+        .orderBy("report_id", "desc")
+
+        .then(feed => {
+          for (var i = 0; i < feed.length; i++) {
+            feed[i].author = {
+              id: feed[i].user_id,
+              name: feed[i]["name"],
+              team: feed[i]["team_name"]
+            };
+
+            if (feed[i].type === "IMAGE") {
+              const imagePath = feed[i]["image_path"];
+
+              if (
+                process.env.DISABLE_IMGIX === "true" ||
+                _.endsWith(imagePath, "gif")
+              ) {
+                feed[i].url =
+                  GCS_CONFIG.baseUrl +
+                  "/" +
+                  GCS_CONFIG.bucketName +
+                  "/" +
+                  imagePath;
+              } else {
+                feed[i].url =
+                  "https://" +
+                  GCS_CONFIG.bucketName +
+                  ".imgix.net/" +
+                  imagePath +
+                  process.env.IMGIX_QUERY;
+              }
+            }
+          }
+          return _.map([number_of_rows, feed]);
```

Comment:
Does this line do anything?

Reply:
Calling map without an iterator function just returns the array, so the map call here does nothing